util/ntp/ntp.go

```diff
@@ -0,0 +1,134 @@
+package ntp
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math"
+	"sync"
+	"time"
+
+	"github.com/beevik/ntp"
+	"github.com/pactus-project/pactus/util/logger"
+)
+
+const (
+	maxClockOffset = time.Duration(math.MinInt64)
+)
+
+type Checker struct {
+	lock   sync.RWMutex
+	ctx    context.Context
+	cancel func()
+
+	threshold  time.Duration
+	offset     time.Duration
+	interval   time.Duration
+	logger     *logger.SubLogger
+	retryTimes int
```

Comment:
What is the point of retrying? 
Generally speaking, we repeat this check every few minutes, and we don't plan to use it as a "single source of truth." Therefore, we don't need to force retries on failure.